packages/graph/hash_graph/lib/graph/src/datastore/mod.rs

```diff
@@ -154,7 +154,7 @@ impl fmt::Display for DatabaseConnectionInfo {
 
 /// Describes the API of a Datastore implementation
 #[async_trait]
-trait Datastore {
+pub trait Datastore: Clone + Send + Sync + 'static {
```

Comment:
Where do we need those bounds? I'd prefer not to have these bounds here but maybe somewhere else.

It feels unintuitive, to require the `Datastore` to be `Clone`. And if it is `Clone`, I don't see why it needs to be `Sync`.

Reply:
These bounds are mostly to allow the `Datastore` to be attached as data on the web api router. Would you prefer we put the datastore behind a RwLock or equivalent on the web api side?

Reply:
We discussed this outside the thread and concluded we will change this in a follow-up PR. 
The general gist is that we don't want to impose `Clone` on all implementations of `Datastore`, and that we can't assume every implementor is pooled